src/storage/v2/storage.cpp

```diff
@@ -1918,7 +1933,9 @@ utils::BasicResult<Storage::CreateSnapshotError> Storage::CreateSnapshot() {
   std::shared_lock<utils::RWLock> storage_guard(main_lock_);
 
   // Create the transaction used to create the snapshot.
-  auto transaction = CreateTransaction(IsolationLevel::SNAPSHOT_ISOLATION);
+  // We don't care about storage mode level here, since snapshot only reads data
+  // We can only add log about potential dirty reads when creating snapshot
+  auto transaction = CreateTransaction(IsolationLevel::SNAPSHOT_ISOLATION, storage_mode_);
 
   // Create snapshot.
```

Comment:
This might cause some snapshots be inconsistent, e.g.: some dandling edges or not symmetric egdes (one end of the edge contains the edge, while the other doesn't). So I would say this can cause serious issues in my opinion.

Reply:
Because analytics mode is already on the polar side of isolations, I think this is not the place where we need to take care of that. The only thing we can do is disable creating snapshots while other transactions are active in the case of analytics mode.

Reply:
> Yes, it can cause issues. In my opinion, we can document this, and enable users to do how they like here.

I think it is not that simple: basically the users can screw up their dataset:
1. Start to ingest data
2. periodic snapshots starts to kick in and in creates a corrupted snapshot
3. Memgraph crashes
4. User restarts Memgraph and voilà, they have a dataset that is corrupted but they don't know about it.

> The only thing we can do is disable creating snapshots while other transactions are active in the case of analytics mode.

I think that is the easiest solution, however then how can our users use snapshots in analytics mode? Maybe triggering it manually and doing it in a blocking manner? I am not aware of the possible use-cases, so these are just my input for somebody, who can make a decision.

Reply:
I think WRITE LOCK when in IN_MEMORY_ANALYTICAL mode is one solution. I will add that and we can discuss whether it makes sense.